### PR TITLE
[REFACTOR][ARITH] Remove legacy compute_expr.h

### PR DESCRIPTION
**Pull request by tqchen**
Replaces most of the ComptuteReduce using foldl.

### CODE REVIEW

**Comment by tqchen** [comment]
cc @junrushao1994 @jroesch @ZihengJiang

**Comment by tqchen** [comment]
@junrushao1994 yes also also remove the legacy templates

File: include/tvm/tir/op.h [hunk 1]
```diff
@@ -762,6 +778,15 @@ inline PrimExpr make_zero(DataType t) {
   }
   return make_const(t, 0);
 }
+
+template <typename FReduce>
+inline PrimExpr foldl(FReduce freduce, PrimExpr init_value, const Array<PrimExpr>& values) {
```
Comment (junrushao1994) [inline include/tvm/tir/op.h:783]: @tqchen I don't have strong opinion, but is it slightly better to use `Foldl` instead to conform the naming convention?

Comment (tqchen) [reply include/tvm/tir/op.h:783]: most of the functions in this file uses stl_case(as they are part of user facing function), we can discuss whether we want to force all function to conform to the CamelCase later.

Comment (junrushao1994) [reply include/tvm/tir/op.h:783]: Okay, it makes sense :-)

**Pull request merged**